cmd/metal-api/internal/service/image-service.go

```diff
@@ -316,3 +322,62 @@ func (ir imageResource) updateImage(request *restful.Request, response *restful.
 		return
 	}
 }
+
+// networkUsageCollector implements the prometheus collector interface.
+type imageUsageCollector struct {
+	ir *imageResource
+}
+
+var (
+	usedImageDesc = prometheus.NewDesc(
+		"metal_image_used_total",
+		"The total number of machines using a image",
+		[]string{"imageID"}, nil,
+	)
+)
+
+func (iuc imageUsageCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(iuc, ch)
+}
+
+func (iuc imageUsageCollector) Collect(ch chan<- prometheus.Metric) {
+	// FIXME bad workaround to be able to run make spec
```

Comment:
We really should fix that, dunno how though

Reply:
Not sure why this should be so bad?

if we call 	
`imgs, err := iuc.ir.ds.ListImages()`
wouldn't it be good practice anyway to check first for 
`  	if iuc.ir == nil || iuc.ir.ds == nil `
even without the spec problem?